assayist/processor/base.py

```diff
@@ -0,0 +1,184 @@
+        :return: a Build object
+        :rtype: assayist.common.models.content.Build
+        """
+        return content.Build.get_or_create({
+            'id_': build_id,
+            'type_': build_type})[0]
+
+    def get_or_create_rpm_artifact(self, id, name, version, release, arch, checksum):
+        """
+        Create or update an Artifact for this rpm.
+
+        :param str id: the rpm's id
+        :param str name: the rpm's name, eg 'kernel'
+        :param str version: the rpm's version, eg '1'
+        :param str release: the rpm's release, eg '2.el7'
+        :param str arch: the rpm's arch, eg 'x86_64'
+        :param str checksum: the rpm's checksum, eg 'deadbeef1234'
+        :return: an Artifact object
+        :rtype: assayist.common.models.content.Artifact
+        """
+        filename = f'{name}-{version}-{release}.{arch}.rpm'
+        if arch == 'src':
+            type = content.Artifact.TYPES['srpm']
+        else:
+            type = content.Artifact.TYPES['rpm']
+        artifact = content.Artifact.create_or_update({
+            'archive_id': id,
+            'type_': type,
+            'architecture': arch,
+            'checksum': checksum,
+            'filename': filename})[0]
+
+        checksum_node = content.Checksum.create_or_update({
+            'checksum': checksum,
+            'algorithm': content.Checksum.guess_type(checksum),
+            'checksum_source': content.Checksum.CHECKSUM_SOURCES['unsigned']})[0]
+
+        artifact.checksums.connect(checksum_node)
+        return artifact
+
+    def get_or_create_rpm_artifact_from_rpm_info(self, rpm_info):
+        """
+        Create or update an Artifact for this rpm from a dictionary.
+
+        :param str rpm_info: A dictionary of information, like one that comes from brew.
+                             Must contain the fields used in get_or_create_rpm_artifact.
+        :return: an Artifact object
+        :rtype: assayist.common.models.content.Artifact
+        """
+        return self.get_or_create_rpm_artifact(
+            id=rpm_info['id'],
+            name=rpm_info['name'],
+            version=rpm_info['version'],
+            release=rpm_info['release'],
+            arch=rpm_info['arch'],
+            checksum=rpm_info['payloadhash'])
+
+    def get_or_create_archive_artifact(self, archive_id, filename, arch, type, checksum):
+        """
+        Create or update an Artifact for this archive.
+
+        :param str archive_id: the archives's id
+        :param str filename: the archive's filename, eg 'maven.jar'
+        :param str arch: the archive's architecture, eg 'x86_64' or None
+        :param str type: the archive's type, eg 'maven'
+        :param str checksum: the archive's checksum, eg 'deadbeef1234'
+        :return: an Artifact object
+        :rtype: assayist.common.models.content.Artifact
+        """
+        if type == 'image':
+            type = content.Artifact.TYPES['container']
```

Comment:
Same as above

Reply:
Also, please don't overwrite method parameters and use a different variable name instead. That'll  make the code below easier to read.